sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/InMemoryPartitionManager.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    /// <summary>
+    ///   A non-volatile in-memory storage service that keeps track of checkpoints and ownership.  It must
+    ///   be used with a single <see cref="EventProcessor"/> instance.
+    /// </summary>
+    ///
+    public class InMemoryPartitionManager : PartitionManager
+    {
+        /// <summary>The primitive for synchronizing access during ownership claim.</summary>
+        private readonly object OwnershipClaimLock = new object();
+
+        /// <summary>The set of stored checkpoints.  Partition ids are used as keys.</summary>
+        private ConcurrentDictionary<string, Checkpoint> Checkpoints;
+
+        /// <summary>The set of stored ownership.  Partition ids are used as keys.</summary>
+        private ConcurrentDictionary<string, PartitionOwnership> Ownership;
+
+        /// <summary>Logs activities performed by this partition manager.</summary>
+        private Action<string> Logger;
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="InMemoryPartitionManager"/> class.
+        /// </summary>
+        ///
+        /// <param name="logger">Logs activities performed by this partition manager.</param>
+        ///
+        public InMemoryPartitionManager(Action<string> logger = null)
+        {
+            Logger = logger;
+
+            Checkpoints = new ConcurrentDictionary<string, Checkpoint>();
+            Ownership = new ConcurrentDictionary<string, PartitionOwnership>();
+        }
+
+        /// <summary>
+        ///   Retrieves a complete ownership list from the in-memory storage service.
+        /// </summary>
+        ///
+        /// <param name="eventHubName">The name of the specific Event Hub the ownership are associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group the ownership are associated with.</param>
+        ///
+        /// <returns>An enumerable containing all the existing ownership for the associated Event Hub and consumer group.</returns>
+        ///
+        public override Task<IEnumerable<PartitionOwnership>> ListOwnershipAsync(string eventHubName,
+                                                                                 string consumerGroup)
+        {
+            return Task.Run(() =>
```

Comment:
May I suggest `Task.FromResult` rather than actually launching a task?

Reply:
Sure. Life is a learning process.